File: CodeTest/src/ReetCode/Easy/1to100/Q66/Q66_PlusOne.java
/**
* Author    : Lee In Bok 
* Date      : 2023.11.02(Thu)
* Spend Time: 02m 04s
* Runtime   : 0 ms (Beats 100%)
* Memory    : 40.4 MB (Beats 97.35%)
* Algoritm  : String
 */

public class Q66_PlusOne {
    public static void main(String[] args) {
        Solution s = new Solution();
        int[] result = s.plusOne(new int[]{9,9,9});

        for(int a : result){
            System.out.print(a + " ");
        }
    }
}

class Solution {
    public int[] plusOne(int[] digits) {
        for(int idx = digits.length - 1; idx >= 0; idx--){
            if(digits[idx] < 9){
                digits[idx]++;
                return digits;
            } 
            
            digits[idx] = 0;
        }

        int[] newArr = new int[digits.length + 1];
        newArr[0] = 1;

        return newArr;
    }
}

File: CodeTest/src/ReetCode/Easy/1to100/Q66/Q66_Plus one.py
"""
# Author    : Lee In Bok 
# Date      : 2023.11.02(Thu)
# Spend Time: 08m 13s
# Runtime   : 30 ms (Beats 97.37%)
# Memory    : 16.2 MB (Beats 38.4%)
# Algoritm  : String
"""

from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        num = (int) (''.join(str(e) for e in digits)) + 1

        return list(map(int, str(num)))

s = Solution()
result = s.plusOne([1,2,3])

print(result)